### PR TITLE
AAP-8667 Fix and improve the JVM management

### PR DESCRIPTION
**Pull request by Alex-Izquierdo**
- Fix how the java version is acquired. 
- Ensure a valid JVM is installed
- Try to detect the default JVM install and set JAVA_HOME if necessary to improve the user experience. 
- Update docs accordingly

Resolves https://issues.redhat.com/browse/AAP-8667

### CODE REVIEW

**Comment by Alex-Izquierdo** [comment]
I can not add you as a reviewer @jamesmarshall24

File: ansible_rulebook/cli.py [hunk 1]
```diff
@@ -46,16 +46,21 @@
...
+import ansible_rulebook.util as util
 
-if os.environ.get("JAVA_HOME") is None:
-    print("JAVA_HOME is not set. " "Please install Java 11+ and set JAVA_HOME")
-    sys.exit(1)
+# ensure a valid JVM is available and configures JAVA_HOME if necessary
+# must be done before importing any other modules
+util.check_jvm()
+if not os.environ.get("JAVA_HOME"):
+    os.environ["JAVA_HOME"] = util.get_java_home()
```
Comment (cutwater) [inline ansible_rulebook/cli.py:55]: Is there a reason why this check must be performed before other imports?

File: ansible_rulebook/cli.py [hunk 2]
```diff
@@ -140,14 +145,12 @@ def get_parser() -> argparse.ArgumentParser:
...
-    java_home = f"{os.environ.get('JAVA_HOME', 'JAVA_HOME not present')}"
-
     result = [
         f"{ansible_rulebook.__version__}",
         f"  Executable location = {sys.argv[0]}",
         f"  Drools_jpy version = {importlib.metadata.version('drools_jpy')}",
-        f"  Java home = {java_home}",
-        f"  Java version = {get_java_version()}",
+        f"  Java home = {util.get_java_home()}",
+        f"  Java version = {util.get_java_version()}",
```
Comment (cutwater) [inline ansible_rulebook/cli.py:153]: Minor: I think calling functions inside f-strings hurts readability.

File: ansible_rulebook/cli.py (hunk 1 above)
Comment (Alex-Izquierdo) [reply ansible_rulebook/cli.py:55]: The other imports load the drools stuff, in that point, you need the JVM

File: ansible_rulebook/cli.py (hunk 2 above)
Comment (Alex-Izquierdo) [reply ansible_rulebook/cli.py]: I buy it. Changed

File: ansible_rulebook/cli.py (hunk 1 above)
Comment (cutwater) [reply ansible_rulebook/cli.py:55]: I'm struggling to find it. Can you please point where the drools stuff is loaded?

Comment (Alex-Izquierdo) [reply ansible_rulebook/cli.py:55]: I think is https://github.com/ansible/ansible-rulebook/blob/main/ansible_rulebook/engine.py#L24 -> https://github.com/ansible/drools_jpy/blob/main/src/drools/ruleset.py

Correct me if I'm wrong @mkanoor 

This was already discovered some time ago, I am not adding it now.

**Review by cutwater — APPROVED** [review]

**Review by mkanoor — APPROVED** [review]

**Pull request merged**